observer/probers/http/http_conf.go

```diff
@@ -54,10 +56,17 @@ func (c HTTPConf) validateRCodes() error {
 	return nil
 }
 
+func (c HTTPConf) validateCollectors(colls map[string]*prometheus.Collector) error {
+	if colls != nil {
+		return errors.New("http prober defines no metrics but received non-nil collector map")
+	}
+	return nil
+}
+
```

Comment:
Since we don't need to implement Instrument() for the HTTPConf, this method can go.